worker/lease/dead_manager_test.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package lease_test
+
+import (
+	"reflect"
+
+	gc "gopkg.in/check.v1"
+
+	lease "github.com/juju/juju/worker/lease"
+)
+
+type DeadManagerSuite struct {
+}
+
+var _ = gc.Suite(&DeadManagerSuite{})
+
+type DeadManagerError struct{}
+
+const DeadManagerErrorMessage = "lease manager stopped"
```

Comment:
There's a couple of ways errors are returned from this worker - the worker loop and individual methods on the manager. The worker loop error is what the runner which controller the worker responds to. The other error is used to inform users of the manager that the manager is stopping - it is stopping by virtue of the fact that the worker loop returns an error. This error triggers the worker running to tell the worker to abort. 

The correct way to test this is to call the methods on the lease Claimer and other interfaces implemented by a manager (in this case the DeadManager) and check that these business methods indeed return "lease manager stopped". Previously such methods would have paniced. 

Then, the other test is to stop the worker and check that the error returned is the same as the one passed in to NewDeadWatcher